src/features/ProfilePage/components/PublicInfo.tsx

```diff
@@ -2,34 +2,101 @@ import styled from 'styled-components';
+
+import type { Mentor, User } from '@/features/Authentication/authenticationApi';
 
 const PublicInfo = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
-
-  const [displayName, setDisplayName] = useState('');
-  const [birthYear, setBirthYear] = useState('');
-  const [area, setArea] = useState('');
-  const [status, setStatus] = useState('');
-  const [isActive, setIsActive] = useState(true);
-  const [story, setStory] = useState('');
+  const dispatch = useAppDispatch();
+  const user = useAppSelector(selectUser);
+  const mentor = useAppSelector(selectMentor);
+  const [updateUser, { isError: isUserError, isSuccess: isUserSuccess }] =
+    useUpdateUserMutation();
+  const [updateMentor, { isError: isMentorError, isSuccess: isMentorSuccess }] =
+    useUpdateMentorMutation();
+
+  const [birthYear, setBirthYear] = useState(String(mentor.birth_year));
+  const [displayName, setDisplayName] = useState(mentor.display_name);
+  const [isAbsent, setIsAbsent] = useState(mentor.is_vacationing);
+  const [region, setRegion] = useState(mentor.region);
+  const [status, setStatus] = useState(mentor.status_message);
+  const [story, setStory] = useState(mentor.story);
   const [topicSearchValue, setTopicSearchValue] = useState('');
 
-  const toggleIsActive = () => setIsActive(isActive => !isActive);
+  const toggleIsAbsent = () => setIsAbsent(!isAbsent);
+
+  const userToSave: User = {
+    ...user,
+    display_name: displayName,
+  };
+
+  const mentorToSave: Mentor = {
+    ...mentor,
+    birth_year: Number(birthYear),
+    display_name: displayName,
+    is_vacationing: isAbsent,
+    region,
+    status_message: status,
+    story,
+  };
+
+  useEffect(() => {
+    if (isUserError || isMentorError) {
+      // TODO: Show error notification
+    }
+  }, [isUserError, isMentorError]);
+
+  useEffect(() => {
+    if (isUserSuccess) dispatch(setUser(userToSave));
+  }, [isUserSuccess]);
+
+  useEffect(() => {
+    if (isMentorSuccess) {
+      dispatch(setMentor(mentorToSave));
+      // TODO: Show success notification
+    }
+  }, [isMentorSuccess]);
+
+  const saveMentor = useDebounce(
+    () => updateMentor(mentorToSave),
+    SAVE_DELAY_MS,
+  );
+
+  const saveUserAndMentor = useDebounce(() => {
+    updateUser(userToSave);
+    updateMentor(mentorToSave);
+  }, SAVE_DELAY_MS);
 
   useEffect(() => {
-    const { displayName, active } = userInfo;
-    if (displayName !== null) setDisplayName(displayName);
-    if (isActive !== null) setIsActive(active);
-  }, [userInfo]);
+    if (Number(birthYear) !== mentor.birth_year) saveMentor();
+    if (displayName !== mentor.display_name) saveUserAndMentor();
+    if (isAbsent !== mentor.is_vacationing) saveMentor();
+    if (region !== mentor.region) saveMentor();
+    if (status !== mentor.status_message) saveMentor();
+    if (story !== mentor.story) saveMentor();
+  }, [birthYear, displayName, isAbsent, region, status, story]);
```

Comment:
This one is a big no. Whatever we try to do here, lets not do it like this.

Reply:
It's gone now!